### ISSUE TITLE
mvt-android cannot check-adb - Libusb_error_Timeout[-7]

### ISSUE THREAD

**Issue by UN1337KN0WN** [mvt-project/mvt#53]
Hi, 

first and foremost great work.
I cloned the repo and did everything to install it properly. Debug mode is on and connected via usb. I executetd the following command: 

#### mvt-android check-adb

The endresult is the following:  
#### Could not receive data from first 950751bf (timeout 5000ms):LIBUSB_ERROR_TIMEOUT [-7]. 

There is nothing that should block it. I even checked if i can access my phone. 
Any clues here ? 

Thanks in advance

**Comment by goshawk22** [comment]
See this: https://github.com/mvt-project/mvt/issues/22#issuecomment-883453657

**Comment by reedcrif** [comment]
> See this: [#22 (comment)](https://github.com/mvt-project/mvt/issues/22#issuecomment-883453657)

Hi, I followed the workaround but I still have the issue (Samsung S20 & adb on MacOS installed with Homebrew).

`adb devices` shows well my device and some commands works eg. `adb backup com.android.providers.telephony`

Is there another workaround?

**Comment by goshawk22** [comment]
Have you run `adb kill-server` followed by `mvt-android check-adb` without running adb devices in between?

**Comment by reedcrif** [comment]
Yes I did:

```
 ~$ adb kill-server
 ~$ mvt-android check-adb
13:37:39 INFO     [mvt.android.cli] Checking Android through adb bridge
         INFO     [mvt.android.modules.adb.chrome_history] Running module ChromeHistory...
13:37:45 CRITICAL [mvt.android.modules.adb.base] Could not receive data from first ******* (timeout 5000ms):
                  LIBUSB_ERROR_TIMEOUT [-7]
```

With `adb devices`, it works:

```
 ~$ adb kill-server && adb devices
* daemon not running; starting now at tcp:5037
* daemon started successfully
List of devices attached
**********    unauthorized
```

I just did a test with a Samsung s9, on the same machine, and it works. Maybe it's not working with my s20...

What do you think?

**Comment by UN1337KN0WN** [comment]
hi @goshawk22, 

thanks for the quick response. I follwed the instruction and i received the same errror message like @reedcrif. I double checked even my usbguard and did it with root. nothing will work.

**Comment by reedcrif** [comment]
A little more details, between  `INFO     [mvt.android.modules.adb.chrome_history] Running module ChromeHistory...` and `13:37:45 CRITICAL [mvt.android.modules.adb.base] Could not receive data from first ******* (timeout 5000ms):
                   LIBUSB_ERROR_TIMEOUT [-7]`, I authorised the USB debugging (when the fingerprint is displayed)

**Comment by dpapperini** [comment]
I have the same issues of @reedcrif and @UN1337KN0WN and I have a Google Pixel 4.

When I try this command `adb devices` the device response right with this message:
`
* daemon not running; starting now at tcp:5037
* daemon started successfully
List of devices attached
XXXXXXXXXXXXXX	device
`

**Comment by Te-k** [comment]
Do you have access to the phone through `adb shell`? If yes, it is an mvt issue, if not, it is an access right issue or installation problem with adb

**Comment by dpapperini** [comment]
> Do you have access to the phone through `adb shell`? If yes, it is an mvt issue, if not, it is an access right issue or installation problem with adb

Yes I have access to the phone through` adb shell`, below the command ` ls -la`:
```
adb shell
flame:/ $ ls -la
total 64
drwxr-xr-x  24 root   root       4096 2009-01-01 01:00 .
drwxr-xr-x  24 root   root       4096 2009-01-01 01:00 ..
dr-xr-xr-x 156 root   root          0 1970-01-09 13:22 acct
drwxr-xr-x  48 root   root        960 2021-07-24 09:11 apex
lrw-r--r--   1 root   root         11 2009-01-01 01:00 bin -> /system/bin
lrw-r--r--   1 root   root         50 2009-01-01 01:00 bugreports -> /data/user_de/0/com.android.shell/files/bugreports
l?????????   ? ?      ?             ?                ? cache -> ?
drwxr-xr-x   4 root   root          0 1970-01-01 01:00 config
lrw-r--r--   1 root   root         17 2009-01-01 01:00 d -> /sys/kernel/debug
drwxrwx--x  47 system system     4096 2021-07-24 09:11 data
d?????????   ? ?      ?             ?                ? data_mirror
drwxr-xr-x   2 root   root       4096 2009-01-01 01:00 debug_ramdisk
lrw-------   1 root   root         23 2009-01-01 01:00 default.prop -> system/etc/prop.default
drwxr-xr-x  20 root   root       6180 2021-07-24 09:13 dev
lrw-r--r--   1 root   root         11 2009-01-01 01:00 etc -> /system/etc
l?????????   ? ?      ?             ?                ? init -> ?
-?????????   ? ?      ?             ?                ? init.environ.rc
d?????????   ? ?      ?             ?                ? linkerconfig
drwx------   2 root   root      16384 2009-01-01 01:00 lost+found
d?????????   ? ?      ?             ?                ? metadata
drwxr-xr-x  15 root   system      320 1970-01-09 13:22 mnt
drwxr-xr-x   2 root   root       4096 2009-01-01 01:00 odm
drwxr-xr-x   2 root   root       4096 2009-01-01 01:00 oem
l?????????   ? ?      ?             ?                ? persist -> ?
d?????????   ? ?      ?             ?                ? postinstall
dr-xr-xr-x 825 root   root          0 1970-01-01 01:00 proc
drwxr-xr-x  14 root   root       4096 2009-01-01 01:00 product
drwxr-xr-x   3 root   root       4096 2009-01-01 01:00 res
lrw-r--r--   1 root   root         21 2009-01-01 01:00 sdcard -> /storage/self/primary
drwx--x---   4 shell  everybody    80 1970-01-09 13:22 storage
dr-xr-xr-x  12 root   root          0 1970-01-09 13:22 sys
drwxr-xr-x  12 root   root       4096 2009-01-01 01:00 system
drwxr-xr-x  10 root   root       4096 2009-01-01 01:00 system_ext
drwxr-xr-x  17 root   shell      4096 2009-01-01 01:00 vendor
```

**Comment by phil0x2e** [comment]
I've got exactly the same problem with a Samsung S10 device.
Device authorization works and `adb shell` also works.

**Comment by nskobelevs** [comment]
Same issue with the OnePlus Nord and on MacOS.

`adb devices` shows the device
`adb shell` can connect to it
`mvt-android check-backup` works 
but same `LIBUSB_ERROR_TIMEOUT` error when trying to download apks.

**Comment by shouvikdutta1998** [comment]
I was facing the same issue again and again and tired of trying all those listed fix but nothing works for me, you can perform the same what I did:

use this over your terminal:  `sudo apt install libusb-0.1-4`

after install this above version of libusb, again check  `mtv-android check-adb`

Hope this help you!!

**Comment by samyxdev** [comment]
> I was facing the same issue again and again and tired of trying all those listed fix but nothing works for me, you can perform the same what I did:
> 
> use this over your terminal: `sudo apt install libusb-0.1-4`
> 
> after install the above version of libusb, again check `mtv-android check-adb`
> 
> Hope this help you!!

I tried with this alternative version of libusb and it still didn't work. Can the others relate?

**Comment by botherder** [comment]
@samyxdev Unfortunately libusb errors are above our control. There will need to be some proper review and probably report upstream to adb-shell, which is the Python library we are using.
In the meantime you can try to upgrade mvt and use the TCP transport instead. For example:
```adb tcpip 5555
mvt-android check-adb --serial 192.168.1.100:5555```
Replacing the IP with the IP of the phone.
